macros/dmt_unit_test.sql

```diff
@@ -1,42 +1,44 @@
-{%- test unit_test(model, input_mapping, expected_output, name, description, compare_columns, depends_on) -%}
+{%- test unit_test(model, input_mapping, expected_output, name, description, compare_columns, exclude_columns, depends_on) -%}
     {%- set test_sql = dbt_datamocktool.get_unit_test_sql(model, input_mapping, depends_on)|trim -%}
-    {%- set test_report = dbt_datamocktool.test_equality(expected_output, name, compare_model=test_sql, compare_columns=compare_columns) -%}
+    {%- set test_report = dbt_datamocktool.test_equality(expected_output, name, compare_model=test_sql, compare_columns=compare_columns, exclude_columns=exclude_columns) -%}
     {{ test_report }}
 {%- endtest -%}
 
-{% test unit_test_incremental(model, input_mapping, expected_output, name, description, compare_columns, depends_on) %}
+{% test unit_test_incremental(model, input_mapping, expected_output, name, description, compare_columns, exclude_columns, depends_on) %}
     {%- set test_sql = dbt_datamocktool.get_unit_test_incremental_sql(model, input_mapping, depends_on)|trim -%}
-    {%- set test_report = dbt_datamocktool.test_equality(expected_output, name, compare_model=test_sql, compare_columns=compare_columns) -%}
+    {%- set test_report = dbt_datamocktool.test_equality(expected_output, name, compare_model=test_sql, compare_columns=compare_columns, exclude_columns=exclude_columns) -%}
     {{ test_report }}
 {% endtest %}
 
-{%- macro test_equality(model, name, compare_model, compare_columns=None) -%}
+{%- macro test_equality(model, name, compare_model, compare_columns=[], exclude_columns=[]) -%}
 
-    {#-- Prevent querying of db in parsing mode. This works because this macro does not create any new refs. #}
+    -- Prevent querying of db in parsing mode. This works because this macro does not create any new refs.
     {%- if not execute -%}
         {{ return('') }}
     {%- endif -%}
 
-    {#- setup -#}
+    -- Setup
     {%- do dbt_utils._is_relation(model, 'test_equality') -%}
-    {#-
-    If the compare_cols arg is provided, we can run this test without querying the
-    information schema — this allows the model to be an ephemeral model
-    -#}
-    {%- if not compare_columns -%}
-        {%- do dbt_utils._is_ephemeral(model, 'test_equality') -%}
-        {%- set exclude_columns = [] -%}
-    {%- else -%}
+    {%- if compare_columns|length and exclude_columns|length -%}
+        {{ exceptions.raise_compiler_error("You cannot provide both compare_columns and exclude_columns") }}
+    {%- endif -%}
+
+    -- If compare_columns have been provided, we need to query the schema to get the list of columns
+    -- to exclude
+    {%- if compare_columns|length -%}
         {%- set all_columns = adapter.get_columns_in_relation(model) | map(attribute='quoted')  -%}
         {%- set exclude_columns = [] -%}
         {%- for col in all_columns -%}
-            {%- set col = col|replace('"',"") -%}
```

Comment:
was this deleted on purpose?

Reply:
Yes sorry should have called this out - I don't think we need it, I pointed it out on [this](https://github.com/mjirv/dbt-datamocktool/pull/67#discussion_r1150625962) PR. More than happy to keep in if you know the reason for it though? Seems to me we just want to replace backticks

Reply:
Might it be a database-specific thing? Maybe Bigquery puts backticks on column names and eg. Postgresql or other databases puts single quotes? The original code came from Leopold Gabelman https://github.com/mjirv/dbt-datamocktool/commit/2209e58745f337ca85b3bf7c53d69e76ec800909

Reply:
This seems to be what caused #84